broker/volumebroker/server/volume_list.go

```diff
@@ -132,7 +143,17 @@ func (s *Server) getAggregateOnmetalVolume(ctx context.Context, id string) (*Agg
 		return nil, status.Errorf(codes.NotFound, "volume %s not found", id)
 	}
 
-	return s.aggregateOnmetalVolume(onmetalVolume, s.clientGetSecretFunc(ctx))
+	onmetalEncryptionSecret := &corev1.Secret{}
+	if secret := onmetalVolume.Spec.Encryption; secret != nil {
+		if err := s.getManagedAndCreated(ctx, secret.SecretRef.Name, onmetalEncryptionSecret); err != nil {
+			if !apierrors.IsNotFound(err) {
+				return nil, fmt.Errorf("error getting onmetal encryption secret %s: %w", secret.SecretRef.Name, err)
+			}
+			return nil, status.Errorf(codes.NotFound, "encryption secret %s not found", secret.SecretRef.Name)
+		}
+	}
+
+	return s.aggregateOnmetalVolume(onmetalVolume, onmetalEncryptionSecret, s.clientGetSecretFunc(ctx))
```

Comment:
If we follow the same flow in `aggregateOnmetalVolume` as for `accessSecret`, this code block won't be necessary.